src/infrastructure/database/repositories/steam.apps.repository.spec.js

```diff
@@ -0,0 +1,792 @@
+        expect(this.result[1].type).toBe("downloadableContent");
+      });
+    });
+  });
+
+  describe(".updateSteamAppsById updates the values of the matching provided steam apps", function () {
+    describe("If two out of three provided steam apps match", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["steam_apps"]);
+
+        this.databaseClient.insertMany("steam_apps", [
+          {
+            appid: 1,
+            name: "Risk of Strain",
+            type: "unknown",
+            triedVia: [],
+          },
+          {
+            appid: 2,
+            name: "Risk of Stain",
+            type: "unknown",
+            triedVia: [],
+          },
+          {
+            appid: 3,
+            name: "Risk of Sprain",
+            type: "unknown",
+            triedVia: [],
+          },
+        ]);
+
+        const apps = [
+          { appid: 1, name: "Risk of Strain", type: "game", triedVia: "steamWeb" },
+          {
+            appid: 2,
+            name: "Risk of Stain",
+            type: "downloadableContent",
+            triedVia: "steamWeb",
+          },
+        ];
+
+        const steamAppsRepo = new SteamAppsRepository(this.databaseClient);
+
+        await steamAppsRepo.updateSteamAppsById(apps);
+
+        this.result = await this.databaseClient.getAll("steam_apps");
+      });
+
+      afterAll(function () {
+        this.databaseClient.disconnect();
+      });
+
+      it("the resulting array has a length of 3", function () {
+        expect(this.result.length).toBe(3);
+      });
+
+      it("the first array has the correct values", function () {
+        expect(this.result[0].appid).toBe(1);
+        expect(this.result[0].name).toBe("Risk of Strain");
+        expect(this.result[0].type).toBe("game");
+        expect(this.result[0].triedVia).toBe("steamWeb");
+      });
+
+      it("the second array has the correct values", function () {
+        expect(this.result[1].appid).toBe(2);
+        expect(this.result[1].name).toBe("Risk of Stain");
+        expect(this.result[1].type).toBe("downloadableContent");
+        expect(this.result[1].triedVia).toBe("steamWeb");
+      });
+
+      it("the third array has the correct values", function () {
+        expect(this.result[2].appid).toBe(3);
+        expect(this.result[2].name).toBe("Risk of Sprain");
+        expect(this.result[2].type).toBe("unknown");
+        expect(this.result[2].triedVia).toEqual([]);
+      });
+    });
+  });
+
+  describe(".getSteamWebUntriedFilteredSteamApps retrieves a specific set of steamApps from the database.", function () {
+    describe("If three steam apps out of eigth match the filters, and the amount of 2 is provided", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["steam_apps"]);
+
+        this.databaseClient.insertMany("steam_apps", [
```

Comment:
Remove duplication, use a function for the mock data. You repeat the same steps in multiple tests.